From 767f90dfd5a75643e8d4181ad07a3e9bdd9c0aca Mon Sep 17 00:00:00 2001
From: Ivan Murashov <ivan.murashov@lge.com>
Date: Thu, 15 Dec 2022 13:13:40 +0300
Subject: [PATCH] libstdc++: Use abs() from std in tflite::task::core::Category

Otherwise build fails when building with use_custom_libcxx=false.
The error example:
third_party/tflite_support/src/tensorflow_lite_support/cc/task/core/category.h:36:12:
error: using integer absolute value function 'abs' when argument
is of floating point type [-Werror,-Wabsolute-value]

Bug: 957519
---
 .../src/tensorflow_lite_support/cc/task/core/category.h         | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/third_party/tflite_support/src/tensorflow_lite_support/cc/task/core/category.h b/third_party/tflite_support/src/tensorflow_lite_support/cc/task/core/category.h
index 9fadd6a5cfe84..4e4ca3f1fa318 100644
--- a/third_party/tflite_support/src/tensorflow_lite_support/cc/task/core/category.h
+++ b/third_party/tflite_support/src/tensorflow_lite_support/cc/task/core/category.h
@@ -33,7 +33,7 @@ struct Category {
   friend bool operator==(const Category& lhs, const Category& rhs) {
     constexpr const double kScoreTolerance = 1e-6;
     return lhs.class_name == rhs.class_name &&
-           abs((double)(lhs.score - rhs.score)) <= kScoreTolerance;
+           std::abs((double)(lhs.score - rhs.score)) <= kScoreTolerance;
   }
 
   friend bool operator!=(const Category& lhs, const Category& rhs) {
-- 
2.25.1

